code/datums/recipe.dm

```diff
@@ -53,15 +53,28 @@
 		return -1
 	return .
 
-/datum/recipe/proc/check_items(obj/container) //1=precisely, 0=insufficiently, -1=superfluous
+/datum/recipe/proc/check_items(obj/container, list/ignored_items = null) //1=precisely, 0=insufficiently, -1=superfluous
 	if(!items)
 		if(locate(/obj/) in container)
-			return -1
-		else
+			if(ignored_items)	//are we supposed to be ignoring anything?
+				var/unignored = FALSE
+				for(var/obj/O in container)
+					if(!is_type_in_list(O, ignored_items))
+						unignored = TRUE	//found something we aren't ignoring, quit looking
+						break
+				if(unignored)
+					return -1
+				else	//anything we found is something we are ignoring
+					return 1
+			else	//not ignoring anything, and we found something
+				return -1
+		else	//didn't find anything
 			return 1
 	. = 1
 	var/list/checklist = items.Copy()
 	for(var/obj/O in container)
+		if(ignored_items && is_type_in_list(O, ignored_items))	//skip if this is something we are ignoring
+			continue

```

Comment:
Given that there is already handling for `if(!items)`... no.

A recipe with no item "ingredients" will fail if items are present UNLESS all present items are also being ignored. As such, the logic above this point has three possible outcomes: failure because of ANY items present not being ignored, success because of ALL items present being ignored, or success because NO items present.

As that logic is only run on recipes with no item ingredient list, and returns early, adding a similar check later in the proc would result in lines that never would run.

Reply:
What I meant is that you can cull that entire hard-to-follow hullabaloo off and leave it just as the check I mentioned
Additionally, change the assignment at line 74 to:
```
var/list/checklist = items ? items.Copy() : list()
```
For your three cases, this will:
A non-ignored item is present:
    It will pass the `continue` check above, do the `items` check I put above, and fail, returning `-1`. Failure!
No items are present:
    This loop will not run, and `.` will remain at the value of `1`. Since `checklist` is null, the statement at line 86 will not force a return of `-1`, and the proc will return `1`. Success!
Only ignored items are present:
    The check at line 76 will prevent the loop from reaching any line past line 77. This case can then be considered the same as "No items are present". Success!

Reply:
Ah, I misunderstood which section of the code you were trying to condense as the check immediately prior to the line commented on... not the block prior to that which you meant.